### ISSUE TITLE
`libexec/spider -o spider-json` conflicts with system installed `json` package

### ISSUE THREAD

**Issue by LecrisUT** [TACC/Lmod#672]
**Describe the bug**

Trying to run `libexec/spider -o spider-json` fails because it is trying to pull in the system's json package instead of the one in `tools/json.lua`.
```
/usr/bin/lua: /usr/share/lua/5.4/json/encode/output.lua:8: variable 'loadstring' is not declared
stack traceback:
        [C]: in function 'error'
        ./libexec/../tools/strict.lua:36: in metamethod 'index'
        /usr/share/lua/5.4/json/encode/output.lua:8: in main chunk
        [C]: in function 'require'
        /usr/share/lua/5.4/json/encode.lua:12: in main chunk
        [C]: in function 'require'
        /usr/share/lua/5.4/json.lua:6: in main chunk
        [C]: in function 'require'
        ./libexec/spider:402: in local 'func'
        ./libexec/spider:494: in function 'main'
        ./libexec/spider:842: in main chunk
        [C]: in ?
```

The system package that it seems to be trying to import is [`lua-json`](https://fedora.pkgs.org/rawhide/fedora-x86_64/lua-json-1.3.4-4.fc39.noarch.rpm.html)

**Desktop (please complete the following information):**
 - OS: Linux
 - Linux distribution: Fedora 38
 - Lmod Version: 8.7.31
 - For Lmod Versions 8.7.7+, please run "module --miniConfig" and
   include the output here.
<details><summary>module --miniConfig</summary>
```
Name                         Where Set  Default                                         Value
----                         ---------  -------                                         -----
LFS_VERSION                  D          1.6.3                                           1.8.0
LMOD_HAVE_LUA_TERM           C          no                                              yes
LMOD_PACKAGE_PATH            D          nil                                             <empty>
LMOD_PAGER                   C          less                                            /usr/bin/more
LMOD_SITEPACKAGE_LOCATION    Other      /usr/share/lmod/8.7.31/libexec/SitePackage.lua  <srctree>
LMOD_SYSTEM_DEFAULT_MODULES  D          __unknown__                                     <empty>
LMOD_TCLSH                   C          tclsh                                           /usr/bin/tclsh
MODULEPATH_ROOT              E                                                          /usr/share/modulefiles
PATH_TO_LUA                  C          lua                                             /usr/bin/lua
```
</details>

**Comment by rtmclay** [comment]
This is not really an Lmod issue.  You are using a Lua that is newer than 5.1.  Then you are matching it with a Json package that was written for Lua 5.1.  Lua 5.2+ removed support for loadstring and replaced it with load.

However Lmod can provide a fix that will allow for the json that uses the loadstring function.

I have committed a new version to the testing branch that should support the older json package.  Please see if this new version on the testing branch fixes your issue.

**Comment by LecrisUT** [comment]
I will try it out this weekend. I'm not familiar with the lua ecosystem so I'll leave it to you guys if there is a good approach that can be done here. Also note #673 if some changes need to be ported there.

I've also submitted an issue upstream https://github.com/harningt/luajson/issues/50

**Comment by rtmclay** [comment]
I have also merged in #673 into the testing branch as well.  Please test this update when you can.
